ls1intum/Artemis
# `Exam mode`: Show relative extension for the working time of student exams

### Checklist
#### General


#### Client
- [x] I translated all new UI strings into German and English.

### Motivation and Context
Closes #4581.

### Description
It was implemented as described in the linked issue.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student registered in the course

1. Setup an exam. Make sure the exam is only visible at some point in the future so that you can edit the working time.
2. Add all course students to the exam.
3. Go to the details of a student exam. You can edit both the absolute value and the percent extension and the other form fields should update accordingly.
    - Setting the ‘extension’ to a value <0% (minimum -100%) to reduce the student's time is intentionally possible.
    - Only changing the seconds a bit might not change the percentage as it is rounded to the nearest integer.
4. After saving, the working time should be shown in the student exam list as shown in the second screenshot.
5. It also should show up on the exam summary page as shown in the third screenshot.
6. Maybe check for other pages where the student exam duration could be shown to check if I might have missed other places other than the two shown in the screenshots.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
| Component | Branch | Statement | |
| --- | --- | --- | --- |
| `StudentExamDetail` | 60% | 87% | 2x 100% for all new/changed code |
| `StudentExamWorkingTime` | 100% | 100% | |

### Screenshots
Instructors can set the working time directly or as extension in percent:
![Screenshot 2022-01-22 at 09-54-19 Exam Management](https://user-images.githubusercontent.com/64250573/151333573-d7e36c0e-e680-4c8d-b70a-9adff079fd3f.png)
In the student exam overview instructors can see the absolute working time and the relative extension for each student:
![Screenshot 2022-01-22 at 09-54-58 Exam Management](https://user-images.githubusercontent.com/64250573/151333577-f3e711c5-0ac7-4798-a8cc-2e9303c15123.png)
When looking at the exam details, both students and instructors can see the same absolute working time and relative extension:
![Screenshot 2022-01-22 at 09-55-18 Exam Management](https://user-images.githubusercontent.com/64250573/151333582-b0b1b89b-38db-46b1-b57d-df7299e2bd26.png)
